azure-servicebus/azure/servicebus/aio/async_client.py

```diff
@@ -0,0 +1,654 @@
+            self.loop, self._get_host(), topic_name, subscription,
+            shared_access_key_name=self.shared_access_key_name,
+            shared_access_key_value=self.shared_access_key_value,
+            debug=self.debug)
+
+    def list_subscriptions(self, topic_name):
+        """Get an async client for all subscription entities in the topic.
+
+        :param topic_name: The topic to list subscriptions for.
+        :type topic_name: str
+        :returns: list[~azure.servicebus.subscription_client.SubscriptionClient]
+        :raises: ~azure.servicebus.common.errors.ServiceBusConnectionError if the namespace is not found.
+        :raises: ~azure.servicebus.common.errors.ServiceBusResourceNotFound if the topic is not found.
+        """
+        try:
+            subs = self.mgmt_client.list_subscriptions(topic_name)
+        except requests.exceptions.ConnectionError as e:
+            raise ServiceBusConnectionError("Namespace: {} not found".format(self.service_namespace), e)
+        except AzureServiceBusResourceNotFound:
+            raise ServiceBusResourceNotFound("Specificed topic does not exist.")
+        sub_clients = []
+        for sub in subs:
+            sub_clients.append(SubscriptionClient.from_entity(
+                self.loop, self._get_host(), topic_name, sub,
+                shared_access_key_name=self.shared_access_key_name,
+                shared_access_key_value=self.shared_access_key_value,
+                debug=self.debug))
+        return sub_clients
+
+
+class SendClientMixin:
+
+    async def send(self, messages, message_timeout=0, session=None, **kwargs):
+        """Send one or more messages to the current entity. This operation
+        will open a single-use connection, send the supplied messages, and close
+        connection. If the entity requires sessions, a Session ID must be set
+        on each outgoing message.
+        This operation is only available on Python 3.5 and above.
+
+        :param messages: One or more messages to be sent.
+        :type messages: ~azure.servicebus.aio.async_message.Message
+        :param message_timeout: The period in seconds during which the Message must be
+         sent. If the send in not completed in this time it will fail.
+        :type message_timeout: int
+        :param session: An optional Session ID. If supplied, and the ServiceBus endpoint
+         is sessionful, this Session ID will be applied to every outgoing message sent with
+         this Sender. If an individual message already has a Session ID, that will be
+         used instead. If no Session ID is supplied here, nor set on an outgoing
+         message, a ValueError will be raised.
+        :type session: str or ~uuid.Guid
+        :raises: ~azure.servicebus.common.errors.MessageSendFailed
+        """
+        async with self.get_sender(message_timeout=message_timeout, session=session, **kwargs) as sender:
+            if isinstance(messages, list):
```

Comment:
It seems `messages` can be Union[Array[Message], Message]
Then, I'd do test if isinstance Mesage, else assume it's an iterable. Because if I put a set of message, it won't work. Or any type of iterable

Reply:
Yeah - this one is bothering me because I also support sending data directly, like:
```
queue_client.send("message")
queue_client.send(123456789)
```
But a string is also an iterable, so I can't make the distinction on that alone :(
Thoughts?

I could just remove that feature and make a user always explicitly pass in a Message object - at least for the first preview. As that removes some ambiguity.

Reply:
Yeah I've decided to remove the feature for now and require that the user pass through an explicit Message object. It's a non-breaking feature that can be added later based on user feedback - so better of to start simple.